overlord/devicestate/devicemgr.go

```diff
@@ -2874,3 +2875,65 @@ func (m *DeviceManager) checkEncryption(st *state.State, deviceCtx snapstate.Dev
 func (m *DeviceManager) encryptionSupportInfo(model *asserts.Model, tpmMode secboot.TPMProvisionMode, kernelInfo *snap.Info, gadgetInfo *gadget.Info, systemSnapdVersions *install.SystemSnapdVersions) (install.EncryptionSupportInfo, error) {
 	return install.GetEncryptionSupportInfo(model, tpmMode, kernelInfo, gadgetInfo, systemSnapdVersions, m.runFDESetupHook)
 }
+
+type EncryptedDisk struct {
+	UUID       string
+	Role       string
+	LegacyKeys map[string]string
+}
+
+func (disk *EncryptedDisk) GetDevPath() string {
+	return fmt.Sprintf("/dev/disk/by-uuid/%s", disk.UUID)
+}
+
+func (m *DeviceManager) GetEncryptedDisks() ([]*EncryptedDisk, error) {
```

Comment:
can you explain why putting this here and not in fdestate directly?  I'm not particularly against one or the other at the moment by trying to understand

Reply:
We need access to the model and modeenv to resolve mount paths. And I felt that it was were doing the most of that.